src/reportSources/legacyFflogs/eventAdapter/sortStatus.ts

```diff
@@ -0,0 +1,70 @@
+import {getDataArrayBy} from 'data'
+import {getActions} from 'data/ACTIONS'
+import {StatusKey, getStatuses} from 'data/STATUSES'
+import {Event} from 'event'
+import _ from 'lodash'
+import {AdapterStep} from './base'
+
+interface EventNode {
+	event: Event
+	isRoot: boolean
+	after: Event[]
+}
+
+export class SortStatusAdapterStep extends AdapterStep {
+	override postprocess(adaptedEvents: Event[]): Event[] {
+		const sortedEvents: Event[] = []
+		let currentEvents: Event[] = []
+
+		for (const event of adaptedEvents) {
+			if (currentEvents.length === 0 || event.timestamp === currentEvents[0].timestamp) {
+				currentEvents.push(event)
+			} else {
+				sortedEvents.push(...this.sortTimestampBucket(currentEvents))
+				currentEvents = [event]
+			}
+		}
+
+		sortedEvents.push(...this.sortTimestampBucket(currentEvents))
+
+		return sortedEvents
+	}
+
+	private canActionApplyStatus = _.memoize((action: number, status: number) => {
+		const statusKey = _.findKey(getStatuses(this.report), other => status === other.id) as StatusKey | undefined
+
+		// This shouldn't be possible, but bail if we don't find a matching status
+		if (statusKey == null) { return false }
+
+		const applyingActions = getDataArrayBy(getActions(this.report), 'statusesApplied', statusKey)
+
+		return applyingActions.some(other => action === other.id)
+	})
+
+	private sortTimestampBucket(events: Event[]): Event[] {
```

Comment:
yeah there's a few places in this implementation that it would be possible to avoid additional array creation; but as it stands the time burnt isn't high enough to worry about it / bother with reducing readability to do it, i don't think.

Reply:
I'm not too happy with it either. I got rid of one of those extra array creations but I know it's still not optimal. I don't think I quite have the brain wrinkles to improve it further.